src/main/java/at/hannibal2/skyhanni/features/rift/everywhere/PunchcardHighlight.kt

```diff
@@ -0,0 +1,75 @@
+package at.hannibal2.skyhanni.features.rift.everywhere
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.HypixelData
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.events.IslandChangeEvent
+import at.hannibal2.skyhanni.events.LorenzChatEvent
+import at.hannibal2.skyhanni.events.RenderMobColoredEvent
+import at.hannibal2.skyhanni.events.withAlpha
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.EntityUtils.isNPC
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.StringUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.repopatterns.RepoPattern
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import net.minecraft.client.Minecraft
+import net.minecraft.client.entity.EntityPlayerSP
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class PunchcardHighlight {
+    private val config get() = SkyHanniMod.feature.rift.punchcard
+    private val ownIGN get() = Minecraft.getMinecraft().thePlayer.name
+    private var lastRiftServer: String = ""
+
+    private val punchedPattern by RepoPattern.pattern(
+        "rift.punchcard.new",
+        "§5§lPUNCHCARD! §r§eYou punched §r§[\\da-z](?:.*?)?(\\w+)§r§[\\da-z] §r§eand both regained §r§a\\+25ф Rift Time§r§e!"
+    )
+
+    val playerList: MutableSet<String> = mutableSetOf()
```

Comment:
make this list private, rather create a public function resetCommand() to use in the commands class.